### PR TITLE
Issue 2025 42stable - deep archive

### PR DESCRIPTION
**Pull request by JustinKyleJames**
Deep Archive and bug fixes backported to 4-2-stable.

### CODE REVIEW

**Comment by JustinKyleJames** [comment]
Ping to take another look after rebase on issue 2082.

**Review by alanking — APPROVED** [review]
Seems fine to me. Let's wait for one more approval before adding #'s

**Review by korydraughn — APPROVED** [review]
Add the #'s

**Pull request merged**
